privacy-security-explainer.md

```diff
@@ -57,9 +60,182 @@ Rounding, quantization, and fuzzing are three categories of mitigations that mod
+}
+
+//AR button click handler
+function onARClick() {
+  xr.requestSession('immersive-ar').then(onARSessionCreated);
+}
+```
 
-# Appendix A: Proposed IDL
-```webidl
-// TODO: Add any necessary IDL
-```
\ No newline at end of file
+In response, the UA must ensure that:
+* The function was invoked in response to a [user activation](#user-activation)
+* The request originates from a [trustworthy document and origin](#trustworthy-documents-and-origins)
+* The request originates from a document that is [visible and has focus](#visibility-and-focus)
+* The request originates from a document allowed to use the WebXR [feature policy](#feature-policy) as well as the [underlying sensors' feature policies](#underlying-sensors-feature-policies)
+* User intention is well understood, either via [explicit consent](#explicit-consent) or [implicit consent](#implict-consent)
+
+If these requirements are not met, the promise returned from `requestSession()` must reject.
+
+## Poses
+### XRPose
+When based on sensor data, calls to `XRFrame.getPose()` will expose sensitive information that may be misused in a number of ways, including input sniffing, gaze tracking, or fingerprinting.
+
+Developers indicate the desire for `XRPose` data by calling `XRFrame.getPose()`.
+
+```js
+function onSessionRafCallback(XRFrame frame) {
+    let motionControllerPose = frame.getPose(xrSession.inputSources[0], xrReferenceSpace);
+}
+```
+
+Before returning poses based on sensor data, the UA must ensure that:
+* User intention is well understood, either via [explicit consent](#explicit-consent) or [implicit consent](#implict-consent); alternatively, in cases where the user experience is not negatively affected, quantization may be applied* The request originates from the document which owns the `XRFrame`'s `XRSession`
+* The document is [visible and has focus](#visibility-and-focus)
+* The `XRSession.visibility` is set to `visible` 
+* **TODO** address issues [#696](https://github.com/immersive-web/webxr/issues/696) and [#724](https://github.com/immersive-web/webxr/issues/724)
+
+### XRViewerPose
+The primary difference between `XRViewerPose` and `XRPose` is the inclusion of `XRView` information. More than one view may be present for a number of reasons. One example is a headset which will generally have two views, but may have more to accommodate greater than 180 degree field of views. Another example is a CAVE system. In all cases, when more than one view is present and the physical relationship between these views is configurable by the user, the relationship between these views is considered sensitive information as it can be used to fingerprint or profile the user.
+
+Developers indicate the desire for `XRViewerPose` data by calling `XRFrame.getViewerPose()`.
+
+```js
+function onSessionRafCallback(XRFrame frame) {
+    let viewerPose = frame.getViewerPose(xrReferenceSpace);
+}
+```
+
+As a result, when more than one view is present, the UA must additionally ensure that:
```

Comment:
Feels like this should more clearly specify that all the restrictions that apply to XRPose still apply here. It hints at that by saying "additionally", but I feel it's far enough removed from the XRPose context that it's worth making explicit.

Reply:
I agree; it's an inherited class but worth calling out that these requirements don't override the previous ones. I'd also suggest making this an ongoing (not one-time) requirement. Suggest:

```suggestion
For every call to getViewerPose(), before returning poses based on sensor data the UA must ensure that all of the requirements of getPose() are met. Further, when more than one view is present, the UA must ensure that:
```